src/analysis/processing/qgsalgorithmhttprequest.cpp

```diff
@@ -0,0 +1,200 @@
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmhttprequest.h"
+#include "qgsprocessingparameters.h"
+#include "qgis.h"
+
+#include <QNetworkAccessManager>
+#include "qgsnetworkaccessmanager.h"
+#include <QNetworkRequest>
+#include <QNetworkReply>
+#include <QUrl>
+#include <QDesktopServices>
+#include <QUrlQuery>
+#include <QMimeDatabase>
+#include <QEventLoop>
+#include <QTimer>
+
+///@cond PRIVATE
+
+QString QgsHttpRequestAlgorithm::name() const
+{
+  return QStringLiteral( "httprequest" );
+}
+
+QString QgsHttpRequestAlgorithm::displayName() const
+{
+  return tr( "HTTP(S) POST/GET request" );
+}
+
+QString QgsHttpRequestAlgorithm::shortDescription() const
+{
+  return tr( "Performs a HTTP(S) POST/GET request and passes back result codes, errormessage and the data (optionally by temporary file)" );
+}
+
+QStringList QgsHttpRequestAlgorithm::tags() const
+{
+  return tr( "open,url,internet,url,fetch,get,post,request,https,http" ).split( ',' );
+}
+
+QString QgsHttpRequestAlgorithm::group() const
+{
+  return tr( "File tools" );
+}
+
+QString QgsHttpRequestAlgorithm::groupId() const
+{
+  return QStringLiteral( "filetools" );
+}
+
+QString QgsHttpRequestAlgorithm::shortHelpString() const
+{
+  return tr( "This algorithm performs a HTTP(S) POST/GET request and passes back HTTP status code and the reply data.\n"
+             "On error it passes back the error code and the message.\n\n"
+             "As well there is the possibility to write the result into a file and/or consider the algorithm as failed on a network error." );
+}
+
+QgsHttpRequestAlgorithm *QgsHttpRequestAlgorithm::createInstance() const
+{
+  return new QgsHttpRequestAlgorithm();
+}
+
+void QgsHttpRequestAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterString( QStringLiteral( "URL" ), tr( "URL" ), QVariant(), false, false ) );
+
+  std::unique_ptr< QgsProcessingParameterEnum > methodParam = std::make_unique < QgsProcessingParameterEnum > (
+        QStringLiteral( "METHOD" ),
+        QObject::tr( "Method" ),
+        QStringList()
+        << QObject::tr( "GET" )
+        << QObject::tr( "POST" ),
+        false,
+        0
+      );
+  methodParam->setHelp( QObject::tr( "The HTTP method to use for the request" ) );
+  addParameter( methodParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterString > dataParam = std::make_unique < QgsProcessingParameterString >(
+        QStringLiteral( "DATA" ), tr( "Data" ), QVariant(), false, true );
+  dataParam->setHelp( QObject::tr( "The data to add in the body if the request is a POST" ) );
+  addParameter( dataParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > failureParam = std::make_unique < QgsProcessingParameterBoolean >(
+        QStringLiteral( "FAILONERROR" ), tr( "Consider error feedback as failure" ), false );
```

Comment:
Processing convention is underscore separating words:
```suggestion
        QStringLiteral( "FAIL_ON_ERROR" ), tr( "Consider HTTP errors as failures" ), false );
```